gridstatus/decorators.py

```diff
@@ -88,9 +89,34 @@ def wrapped_f(*args, **kwargs):
             if args_dict["date"] == "latest":
                 return f(*args, **kwargs)
 
+            default_timezone = args_dict["self"].default_timezone
+
+            # For today with hourly data, create a range that spans the day
+            if self.frequency == "HOUR_START":
+                date = args_dict.get("date")
+                if date == "today":
+                    args_dict["date"] = pd.Timestamp.now(tz=default_timezone).floor("D")
+                    args_dict["end"] = pd.Timestamp.now(tz=default_timezone).ceil("D")
+                # If date is a string or datetime.date, then we floor it to the
+                # beginning of the date. If the end date is not set, we set it to
+                # the start of the next day
+                elif isinstance(date, str) or isinstance(date, datetime.date):
```

Comment:
if we dont add this condition what breaks exactly? the situation i could imagine is something like where it'd be reasonable to return full day

```python
IESO().get_load(date="2023-12-30")
IESO().get_load(date="2023-12-30 08:00")
``` 

However, it seems like this also handles cases like 

```python
IESO().get_load(date="2023-12-30 08:00", end="2023-12-30 10:00"
```

where it would change the range to be 
```python
IESO().get_load(date="2023-12-30 00:00", end="2023-12-31 00:00"
```

I guess, cant imagine a case where the we need this if a user also supplies an end date.

Reply:
Yeah, I've removed the check so `datetime.datetime` will also hit this branch.

Reply:
why not just always do this for any value of `date`? dont we also want it to apply to `pd.Timestamp`?

Reply:
My reasoning was we should not alter a timestamp for `date` with the assumption the user only wants data from that timestamp on. You have pointed out though that users can do the filtering themselves.

Reply:
I removed the check.